challenge_1/rust/makernaren/src/string_reverser.rs

```diff
@@ -0,0 +1,34 @@
+/*Not an efficient solution, but I learnt arrays, vectors in rust and can't 
+define run time array in rust :( */
+pub fn reverse(input: &str) -> String {
+	let n = input.len();
+	// Define a vector of char type with length of the input string
+	let mut char_vec = vec!['a'; n];
+	// Replace the chars to tail of the vector
+	for (i, c) in input.chars().enumerate() {
+		char_vec[n-1-i] = c
+		// char_array.push(c)
+	}
+	// Iterate, copy and collect it as a string
+	let result: String = char_vec.iter().cloned().collect();
+	return result
+	// oneliner : return input.chars().rev().collect::<String>();
```

Comment:
Rust allows you to just leave the expression of desired output on the last line of a function/block's scope so that return is only necessary if explicitly done before the end of a function/block.

For example, instead of:
```rust
let result: String = char_vec.iter().cloned().collect();
return result
```
...you could write something like:
```rust
char_vec.iter().cloned().collect::<String>()
```